File: greedy/cpp/b4796.cpp
#include <iostream>

int main() {
	int tc = 1;
	while (true) {
		int L, P, V;
		std::cin >> L >> P >> V;
		if (L == 0 && P == 0 && V == 0) break;

		int division, remainder;
		division = V / P;
		remainder = V % P;
		remainder = L < remainder ? L : remainder;
		int result = division * L + remainder;
		std::cout << "Case " << tc << ": " << result << "\n";
		tc += 1;
	}
}

File: greedy/python/b4796.py
from sys import stdin

tc = 1
while True:
    L, P, V = map(int, stdin.readline().split())
    if L == 0 and P == 0 and V == 0:
        break
    division = V // P
    remainder = V % P
    remainder = L if L < remainder else remainder
    result = division * L + remainder
    print("Case {}: {}".format(tc, result))
    tc += 1